network/src/peer_manager/mod.rs

```diff
@@ -447,9 +451,27 @@ where
                     "{} New connection established: {}", self.network_context, conn.metadata
                 );
 
-                // Update libra_network_peer counter.
-                self.add_peer(conn);
-                self.update_connected_peers_metrics();
+                // TODO: Keep track of somewhere else to not take this hit in case of DDoS
+                let inbound_conns = self
+                    .active_peers
+                    .iter()
+                    .filter(|(_, (metadata, _))| metadata.origin == ConnectionOrigin::Inbound)
+                    .count();
+
+                // Reject excessive inbound connections by letting them just drop out of scope
+                // We control outbound connections with Connectivity manager before we even send them
+                // and we must allow connections that already exist to pass through tie breaking.
+                // TODO: Allow for trusted peers to still connect
+                if conn.metadata.origin == ConnectionOrigin::Outbound
+                    || self
+                        .active_peers
+                        .contains_key(&conn.metadata.remote_peer_id)
+                    || inbound_conns < self.inbound_connection_limit
+                {
+                    // Add new peer, updating counters and all
+                    self.add_peer(conn);
+                    self.update_connected_peers_metrics();
+                }
```

Comment:
Let's move the "New connection established" log into the if branch and add a log to the else branch that tells us the connection was dropped b/c we're at the inbound connection limit.

Reply:
Now, realized we aren't really closing the connection via the socket, so I'll add that.